src/NUnitFramework/framework/Internal/Execution/ParallelWorkItemDispatcher.cs

```diff
@@ -242,6 +242,11 @@ private void RestoreQueues()
 
         private void OnEndOfShift(object sender, EventArgs ea)
         {
+            // This shift ended, so see if there is work in any other
+            // TODO: Temp fix - revisit later when object model is redesigned.
+            if (StartNextShift())
+                return;
+
```

Comment:
This change is an "extra" not related to the issue I'm fixing. But it's an obvious error. Without it, we would restore all the queues, possibly losing work items not yet started, when the shift starts. For example, if a NP fixture has parallel tests, they could be waiting in the parallel queue to be run when the non-p shift ends!

More work is needed here - I just fixed the obvious.